src/RMPI/onesided.jl

```diff
@@ -10,11 +10,12 @@ function mpi_one_sided(data, comm = mpi_comm(), root = mpi_root)
     MPI.Initialized() || error("MPI needs to be initialised first.")
     np = MPI.Comm_size(comm)
     id = MPI.Comm_rank(comm)
-    P = pairtype(data)
+    P = eltype(data)
 
     # compute the required capacity for the communication buffer as a
     # fraction of the capacity of `data`
-    cap = capacity(data) ÷ np + 1
+    # TODO: ugly hack
+    cap = length(storage(data).keys) ÷ np + 1
```

Comment:
We discussed this earlier. I think `capacity` should be passed as an argument to `mpi_one_sided()`. I don't think that we can safely anticipate the required buffer size any more from looking at `data`.